vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -155,6 +162,17 @@ def parse_pod(pod)
         end
       end
 
+      createdby_txt = pod.metadata.annotations.try("kubernetes.io/created-by")
```

Comment:
@Fryguy @jrafanie @tenderlove , I think what you are looking for is actually this:
https://github.com/aetherknight/recursive-open-struct/blob/master/lib/recursive_open_struct.rb#L41

Kubeclient relies on RecursiveOpenStruct aka ROS, so in fact that's just a hash behind the scenes based on the received json (well, nested hash)
You can access it in a "hashy" way e.g. 
node["metadata"]["annotations"]
or leverage the fact that ROS defines methods dynamically for the keys when the hash is built and allow a 'method like access':
node.metadata.annotations
The reason @simon3z does "try" there is because the annotations entry doesn't always exist, and since he tries to access something nested under annotations, he is being careful :)

Even if he would have gone with hash regular access approach, since he is checking for a nested element existence, this would of course fail anyway with "undefined method `[]' for nil:NilClass"
(for pod["mynewsection"]["annotations"] where the key "mynewsection" doesn't exist)

Now, annotations are something relatively new and we didn't deal with them before. 
As I examine some annotations now, the names that are used there with dots and slashes such as "kubectl.kubernetes.io/desired-replicas" are gonna make this 'method like access' more fragile and harder - dot in the middle of the key will be interpreted as another method call and fail.
(IMO, even for a key name in hash something like "kubectl.kubernetes.io/desired-replicas" is a stretch, but I don't assign those key names, we just consume them)  

So we can decide that for annotations we could go with what's done above, or with just plain hash access with square brackets for clarity for "future us". and with the "try" of course. thoughts?
I hope this clarifies the questions raised in this thread.